src/hooks/use-deck-state.ts

```diff
@@ -37,60 +37,65 @@ type ReducerActions =
   | { type: 'COMMIT_TRANSITION'; payload?: DeckView }
   | { type: 'CANCEL_TRANSITION'; payload?: undefined };
 
+// If payload is null, we don't want to merge because it nulls out the whole
+// object.
+const mergeView = (view: DeckView, payload: Partial<DeckView>) =>
+  payload ? merge(view, payload) : view;
+
 function deckReducer(state: DeckState, { type, payload = {} }: ReducerActions) {
   switch (type) {
```

Comment:
Perhaps it would be more semantic, and easier to debug / remove later, if we just added another default like we do for `undefined`?
```
function deckReducer(state: DeckState, { type, payload = {} }: ReducerActions) {
  // Safeguard against nulls:
  if (!payload) {
    payload = {};
  }
  switch (type) {
```